WordPress/gutenberg
# Patterns: Add user pattern categories to post editor inserter patterns tab

## What?
Add user pattern categories to the post editor block inserter panel

## Why?
With the addition of the ability for users to categorize their patterns they will expect them to be ordered  by those categories in the inserter.

## How?
Used the new `__experimentalBlockPatternCategories` block editor setting

This PR was part of a group of PRs that were merged back into https://github.com/WordPress/gutenberg/pull/53835 before being merged into trunk.

A full list of the PRs involved in this work can be [found here](https://github.com/WordPress/gutenberg/issues/53164).

## Testing Instructions

- Add some unsynced patterns with categories assigned (currently on unsynced patterns display in the patterns category panel - follow up PRs will merge synced patterns into this)
- Click on the inserter Patterns panel and make sure the patterns appear on the correct category and can be inserted
- Also add an uncategorized pattern and make sure it appears in an `Uncategorized` category

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/3629020/f7dd06b6-a677-4117-9f90-6e6f97bce0aa
